refactor/project-ui

ahmedNaser7 opened the pull request:
feat: implement snackbar for error messages in MessageDetailsScreen and
 add whatsapp phone text to make use know which otp type will send 

introduces a snackbar to display error messages in the `MessageDetailsScreen`.

The following changes were made:
- Added `onDismissSnackBar` to `MessageInteractionListener`.
- Added `showSnackBar` boolean to `MessageDetailsUiState`.
- Implemented `onDismissSnackBar` in `MessagesDetailsViewModel` to update the UI state.
- Modified `MessageDetailsScreen` to display a `SnackBar` component when `showSnackBar` is true and an error message exists.
- Updated error handling in `MessagesDetailsViewModel` to set `showSnackBar` to true when an error occurs during message sending.
- The `LostConnectionScreen` is now only shown if `showSnackBar` is false, preventing both from being displayed simultaneously.

ahmedNaser7 committed:
Clarify phone number type and center text in RegisterScreen

This commit updates the string resource for "enter_your_phone_number_to_continue" to specify "whatsapp phone number" for clarity.

Additionally, the text alignment for this message in the RegisterScreen is now centered.

ahmedNaser7 committed:
feat: implement snackbar for error messages in MessageDetailsScreen

This commit introduces a snackbar to display error messages in the `MessageDetailsScreen`.

The following changes were made:
- Added `onDismissSnackBar` to `MessageInteractionListener`.
- Added `showSnackBar` boolean to `MessageDetailsUiState`.
- Implemented `onDismissSnackBar` in `MessagesDetailsViewModel` to update the UI state.
- Modified `MessageDetailsScreen` to display a `SnackBar` component when `showSnackBar` is true and an error message exists.
- Updated error handling in `MessagesDetailsViewModel` to set `showSnackBar` to true when an error occurs during message sending.
- The `LostConnectionScreen` is now only shown if `showSnackBar` is false, preventing both from being displayed simultaneously.

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/san3a/presentation/screen/messagesDetails/MessageDetailsScreen.kt:
```diff
@@ -107,7 +110,7 @@ fun MessageDetailsContent(
         ) {
             when {
-                state.errorMessage != null -> {
+                state.errorMessage != null && state.showSnackBar.not() -> {
```
it should has no relation with showSnackBar

Joseph-Sameh-0 reviewed (commented):
the snack bar should disappear after 2 sec

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/san3a/presentation/screen/messagesDetails/MessagesDetailsViewModel.kt:
```diff
@@ -151,7 +151,8 @@ class MessagesDetailsViewModel(
                     screenState.value.copy(
                         errorMessage = it.message,
-                        sendButtonState = AppButtonState.Enable
+                        sendButtonState = AppButtonState.Enable,
+                        showSnackBar = true
```
on send text messages and also in image message

Joseph-Sameh-0 replied:
and update the state by sendingTextMessage  = null in the on error case

ahmedNaser7 replied:
Okay i will make it

Joseph-Sameh-0 committed:
Refactor: Enhance error handling in MessagesDetailsViewModel

This commit introduces several improvements to error handling within the `MessagesDetailsViewModel`:

- **Message Count Correction:** When sending a text or image message fails, the `messagesSize` is now correctly decremented to reflect the failed send attempt.
- **Clear Sending State on Error:** If sending a text or image message fails, the `sendingTextMessage` or `sendingImageMessage` state is reset to `null`.
- **Automatic Snackbar Dismissal:** A new `hideSnackBar` function has been added. This function is called after an error occurs, automatically hiding the error message snackbar after a 3-second delay. This improves the user experience by preventing the snackbar from persisting indefinitely.

Joseph-Sameh-0 commented:
Localize the error and make it of type UiText

ahmedNaser7 committed:
Fix: Make SnackBar message a string resource

The SnackBar message in MessageDetailsScreen was previously using a UiText object directly from the state. This change updates it to use a string resource `R.string.some_error_happened` for better localization and consistency.

ahmedNaser7 committed:
Fix: improve error handling in messages screen

This commit enhances error handling in the messages screen by:

- Changing `errorMessage` in `MessageDetailsUiState` from `String?` to `UiText?` to support localized error messages.
- Updating `MessagesDetailsViewModel` to use `UiText.StringResource` for error messages, providing localized feedback.
- Displaying the localized error message in the `SnackBar` within `MessageDetailsScreen`.
- Adding a new string resource `occur_error_while_using_messages_screen` for English and Arabic.
